aptos-move/e2e-move-tests/src/tests/module_event.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{assert_success, assert_vm_status, tests::common, MoveHarness};
+use aptos_package_builder::PackageBuilder;
+use aptos_types::{account_address::AccountAddress, on_chain_config::FeatureFlag};
+use move_core_types::{language_storage::TypeTag, vm_status::StatusCode};
+use serde::{Deserialize, Serialize};
+use std::str::FromStr;
+
+#[derive(Debug, Serialize, Deserialize, Eq, PartialEq)]
+struct Field {
+    field: bool,
+}
+
+#[derive(Debug, Serialize, Deserialize, Eq, PartialEq)]
+struct MyEvent {
+    seq: u64,
+    field: Field,
+    bytes: Vec<u64>,
+}
+
+#[test]
```

Comment:
Can we also tests for more cases such as:
1. Publishing modules with event emit calls with non-event structs.
2. Events with nested types (phantom vs not)
3. etc.

Reply:
1. The compilation will not go through.